File: rft/Models/Exam.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace rft.Models
{
    public class Exam
    {
        public int Id { get; set; }
        [ForeignKey("User")]
        public int MakerID { get; set; }
        public DateTime DateTime { get; set; }
        public string Course { get; set; } // course name of the exam
        public string Location { get; set; } // room
    }
}

File: rtf_ui/lib/models/exam.dart
class Exam {
  int id;
  int makerID;
  String maker = "";
  DateTime dateTime;
  String course;
  String location;
  Exam({
    required this.id,
    required this.makerID,
    required this.dateTime,
    required this.course,
    required this.location,
  });
  factory Exam.fromJson(Map<String, dynamic> json) {
    return Exam(
      id: json['id'],
      course: json['course'],
      dateTime: DateTime.parse(json['dateTime']),
      location: json['location'],
      makerID: json['makerID'],
    );
  }
}
